### PR TITLE
Adding tag descriptions to tooltips

### PR DESCRIPTION
**Pull request by davidsharp7**
### Problem

Currently tags displayed at the column level show the tag but not the description which can help the end user understand what the tag means

Closes: #2611

### Solution

Get the tag descriptions from the tags end point and when a column has a tag display the corresponding description on hover over.(https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

One-line summary:

add hover over to each column level tag which displays the tag description.

Screenshot 2023-09-12 at 15 41 20

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [x] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [x] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

File: web/src/components/datasets/DatasetInfo.tsx [hunk 1]
```diff
@@ -42,28 +43,35 @@ type DatasetInfoProps = {
...
+  
 
 const DatasetInfo: FunctionComponent<DatasetInfoProps> = (props) => {
   const { datasetFields, facets, run, jobFacets, fetchJobFacets, resetFacets } = props
   const i18next = require('i18next')
 
   useEffect(() => {
     run && fetchJobFacets(run.id)
+    fetchTags()
   }, [])
```
Comment (phixMe) [inline web/src/components/datasets/DatasetInfo.tsx:75]: This depends on the `run` You'll want to rerun this if the run changes.

File: web/src/store/actionCreators/actionTypes.ts [hunk 2]
```diff
@@ -57,3 +57,7 @@ export const FETCH_RUN_FACETS = 'FETCH_RUN_FACETS'
 export const FETCH_JOB_FACETS = 'FETCH_JOB_FACETS'
 export const FETCH_FACETS_SUCCESS = 'FETCH_FACETS_SUCCESS'
 export const RESET_FACETS = 'RESET_FACETS'
+
+// tags
+export const FETCH_TAGS = 'FETCH_TAGS'
+export const FETCH_TAGS_SUCCESS = 'FETCH_TAGS_SUCCESS'
```
Comment (phixMe) [inline web/src/store/actionCreators/actionTypes.ts:63]: I think you've got a few newline issues. The project has a formatter that should help you in your IDE.

File: web/src/components/datasets/DatasetInfo.tsx [hunk 3]
```diff
@@ -63,7 +72,7 @@ const DatasetInfo: FunctionComponent<DatasetInfoProps> = (props) => {
   const i18next = require('i18next')
 
   useEffect(() => {
-    run && fetchJobFacets(run.id)
+    run && fetchJobFacets(run.id) && fetchTags()
   }, [])
```
Comment (phixMe) [inline web/src/components/datasets/DatasetInfo.tsx:76]: I think you would need to add `[run]` to the dependencies list. An empty list runs on the component mount. I think it should work for you.

Comment (davidsharp7) [reply web/src/components/datasets/DatasetInfo.tsx:76]: yeah sorry misunderstood what you were saying - rookie error!

**Review by phixMe — APPROVED** [review]

**Pull request merged**